### PR TITLE
Parameters design

### PR DESCRIPTION
**Pull request by nilebox**

### CODE REVIEW

File: docs/parameters.md [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+Parameter values could be passed as JSON objects, the same way as for [individual parameters]((#passing-parameter-value-as-a-json-object)):
+```yaml
+  ...
+  parametersFrom:
+    - type: json
+      secretRef:
+        name: mysecret
+```
+
+### Conflict resolution
```
Comment (ash2k) [inline docs/parameters.md]: I'd add a way to specify how conflicts are handled. What you described could be last-one-wins. Also `Fail` is an option when you don't expect conflicts and don't want surprises.

File: docs/parameters.md [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+  "service_id":"service-guid-111",
+  "plan_id":"plan-guid",
+  "organization_guid": "org-guid",
+  "space_guid":"space-guid",
+  "parameters": {"ami_id":"ami-ecb68a84"}
+}' -H "Content-Type: application/json"
+```
+
+Note that broker accepts `ami_id` parameter ([AMI](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AMIs.html) identifier).
+To configure a provisioning request in Service Catalog, we need to declare an `Instance` resource with specifying AMI
```
Comment (TimCinel) [inline docs/parameters.md]: Nitpick

s/with specifying AMI identifier/with a specified AMI identifier
s/with specifying AMI identifier/with an AMI identifier specified

File: docs/parameters.md [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+  - [Passing parameter value as a JSON object](#passing-parameter-value-as-a-json-object)
+  - [Passing all parameters from a single source at once](#passing-all-parameters-from-a-single-source-at-once)
+  - [Conflict resolution](#conflict-resolution)
+
+## Overview
+`parameters` and `parametersFrom` properties of `Instance` and `Broker` resources provide support for passing 
+parameters to the broker relevant to the corresponding
+[provisioning](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#provisioning) or
+[binding](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#binding) request. 
+The resulting structure represents an arbitrary JSON object, which is assumed to be valid for a particular broker.
```
Comment (nilebox) [inline docs/parameters.md]: Clarify: s/resulting structure/Service Catalog processes parameters and produces an arbitrary JSON object ....

File: docs/parameters.md [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+  serviceClassName: aws-ami
+  planName: default
+  parameters:
+    - name: ami_id
+      value: ami-ecb68a84
+```
+
+### Passing sensitive data
+
+To pass a parameter value stored in a `Secret` or a `ConfigMap`, use the `valueFrom` field:
```
Comment (nilebox) [inline docs/parameters.md]: parametersFrom

File: docs/parameters.md [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+  name: qwerty-instance
+  namespace: test-ns
+spec:
+  serviceClassName: qwerty
+  planName: default
+  parameters:
+    - name: txt
+      value: Plain text, passed as a string
+    - name: meta
+      type: json # Explicitly tells that the value is JSON which should be validated and passed as a JSON object
```
Comment (nilebox) [inline docs/parameters.md]: rephrase: tells Service Catalog to treat the value as a JSON blob....

File: docs/parameters.md [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+                {
+                  "type": "b",
+                  "B": "2",
+                },
+            ],
+            "labels": ["foo", "bar"]
+          }
+```
+
+If the JSON object is stored in a `Secret` or a `ConfigMap` key, the combination of `valueFrom` and `type: json` fields
```
Comment (nilebox) [inline docs/parameters.md]: parametersFrom

File: docs/parameters.md [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+  serviceClassName: qwerty
+  planName: default
+  parameters:
+    - name: txt
+      value: Plain text, passed as a string
+    - name: meta
+      type: json # Explicitly tells that the value is JSON which should be validated and passed as a JSON object
+      value: >
+          {
+            "attributes": [
```
Comment (nilebox) [inline docs/parameters.md]: justify why we need support for JSON objects

**Pull request closed without merging**